### PR TITLE
Move tests from libethereum

### PR DESCRIPTION
**Pull request by winsvega**

### CODE REVIEW

**Comment by LefterisJP** [comment]
How is this supposed to build? You are introducing a dependency to libraries in libethereum when building.

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
Eth_ETHASH-CL_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_ETHASH_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_ETHCORE_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_ETHEREUM_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_EVMASM_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_EVMCORE_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_EVM_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_LLL_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test
Eth_TESTUTILS_LIBRARY
    linked by target "testweb3core" in directory /var/jenkins_home/sharedspace/libweb3core/label/master/libweb3core/test

-- Configuring incomplete, errors occurred!
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,3 +31,4 @@ if (NOT EMSCRIPTEN)
 	endif()
 endif()
 
+add_subdirectory(test)
```
Comment (arkpar) [inline CMakeLists.txt]: Please add a configuration option to build without tests:

```
if (TESTS)
    add_subdirectory(test)
endif()
```

**Comment by arkpar** [comment]
Files should be moved to a new repo with history preserved. See http://gbayer.com/development/moving-files-from-one-git-repository-to-another-preserving-history/

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+add_subdirectory(libdevcore)
+add_subdirectory(libdevcrypto)
+add_subdirectory(libp2p)
+
+set(SRC_LIST ${SRC_LIST} ${SRCTEST})
+
+file(GLOB HEADERS "*.h")
+add_executable(testweb3core ${SRC_LIST} ${HEADERS})
+eth_use(testweb3core REQUIRED Eth::ethereum Eth::testutils)
```
Comment (arkpar) [inline test/CMakeLists.txt]: libweb3core tests can't depend on Eth::ethereum and Eth::testutils

Comment (LefterisJP) [inline test/CMakeLists.txt]: As an additional note on that,  If Eth::testutils is supposed to be used by tests in more than one repos then they should be moved to libweb3core.

Comment (winsvega) [inline test/CMakeLists.txt]: how it could use eth objects then? 
such as eth::Transaction ?

Comment (LefterisJP) [inline test/CMakeLists.txt]: If it's a test in `libweb3core` it should not. Only test in libethereum should use those.

**Pull request closed without merging**